server/repository/src/db_diesel/item_link_row.rs

```diff
@@ -0,0 +1,98 @@
+use super::{item_link_row::item_link::dsl::*, item_row::item, StorageConnection};
+
+use crate::repository_error::RepositoryError;
+
+use diesel::prelude::*;
+
+table! {
+    item_link (id) {
+        id -> Text,
+        item_id -> Text,
+    }
+}
+
+joinable!(item_link -> item (item_id));
+
+#[derive(Clone, Insertable, Queryable, Debug, PartialEq, AsChangeset, Eq)]
+#[table_name = "item_link"]
+pub struct ItemLinkRow {
+    pub id: String,
+    pub item_id: String,
+}
+
+impl Default for ItemLinkRow {
+    fn default() -> Self {
+        Self {
+            id: Default::default(),
+            item_id: Default::default(),
+        }
+    }
+}
+
+pub struct ItemLinkRowRepository<'a> {
+    connection: &'a StorageConnection,
+}
+
+impl<'a> ItemLinkRowRepository<'a> {
+    pub fn new(connection: &'a StorageConnection) -> Self {
+        ItemLinkRowRepository { connection }
+    }
+
+    #[cfg(feature = "postgres")]
+    pub fn upsert_one(&self, item_link_row: &ItemLinkRow) -> Result<(), RepositoryError> {
+        diesel::insert_into(item_link)
+            .values(item_link_row)
+            .on_conflict(id)
+            .do_update()
+            .set(item_link_row)
+            .execute(&self.connection.connection)?;
+        Ok(())
+    }
+
+    #[cfg(not(feature = "postgres"))]
+    pub fn upsert_one(&self, item_link_row: &ItemLinkRow) -> Result<(), RepositoryError> {
+        diesel::replace_into(item_link)
```

Comment:
Just a pesonal preference, I've been using, `use item_link:dsl as item_line_dsl`, and use that one rathern them item_link, just a bit more expicit about what time_line is, but is a lot more verbose. Also the overall import could just be use `super::{item_link_row::item_link::dsl:: ..``, -> `use self::item_link::dsl:: ..` (vs importing from super what's been exported here)

Reply:
I agree decided to chuck in, thanks!